redwood-protocol-guest/src/commonMain/kotlin/app/cash/redwood/protocol/guest/ProtocolWidgetChildren.kt

```diff
@@ -37,13 +37,10 @@ public class ProtocolWidgetChildren(
   }
 
   override fun remove(index: Int, count: Int) {
-    for (i in index until index + count) {
-      val widget = _widgets[i]
-      guestAdapter.removeWidget(widget.id)
-      widget.depthFirstWalk(guestAdapter.childrenRemover)
+    for (i in index + count - 1 downTo index) {
+      val widget = widgets[i]
+      guestAdapter.appendRemove(id, tag, index, widget.id)
```

Comment:
Notable changes here:
- We do not actually remove the widget anymore, only record the remove change
- We unconditionally emit single-widget removes